MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -2,7 +2,153 @@ import UIKit
+            title: result.title,
+            message: result.text,
+            preferredStyle: .alert
+        )
+        let action = UIAlertAction(
+            title: result.buttonText,
+            style: .default, handler: {_ in
+            self.currentQuestionIndex = 0
+            self.rightAnswerCount = 0
+            self.showStep(quize: self.createStepModel(model: self.questions[0]))
+        })
+        alert.addAction(action)
+
+        present(alert, animated: true, completion: nil)
+    }
+    
+    private func createStepModel(model: QuizeQuestion) -> QuizeStepViewModel {
+        return QuizeStepViewModel(
+            image: UIImage(named: model.image) ?? UIImage(),
+            question: model.text,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questions.count)"
+        )
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        noButton.isUserInteractionEnabled = false
+        yesButton.isUserInteractionEnabled = false
+        movieImageView.layer.borderWidth = 8
+        movieImageView.layer.borderColor = isCorrect ? UIColor.ypGreen.cgColor : UIColor.ypRed.cgColor
+        if isCorrect {
+            rightAnswerCount += 1
+        }
+    }
+    private func getResultMessage() -> String {
+        let formater = DateFormatter()
+        formater.dateFormat = "dd.MM.yyyy hh:mm"
+        let result: String = "Ваш результат: \(rightAnswerCount)/\(questions.count)."
+        let quize: String = "Количество сыгранных квизов: \(quizeCount)."
+        let record: String = "Рекорд: \(answersRecord)/\(questions.count) (\(formater.string(from: recordDate)))."
+        let percent = (Double(totalRightAnswerCount) / Double(quizeCount * questions.count) * 10000).rounded() / 100
+        let statistic: String = "Средняя точность: \(percent)%."
+        return result + "\n" + quize + "\n" + record + "\n" + statistic
+    }
+    private func showNextQuestionOrResults() {
+        movieImageView.layer.borderWidth = 0
+        movieImageView.layer.borderColor = UIColor.white.withAlphaComponent(0.0).cgColor
+        if currentQuestionIndex == questions.count - 1 {
+            if quizeCount == 0 || rightAnswerCount > answersRecord {
+                answersRecord = rightAnswerCount
+                recordDate = Date()
+            }
+            totalRightAnswerCount += self.rightAnswerCount
+            quizeCount += 1
+            showResult(quize: QuizeResultsViewModel(
+                title: "Этот раунд окончен",
+                text: getResultMessage(),
+                buttonText: "Сыграть еще раз"
+            )
+            )
+        } else {
+            currentQuestionIndex += 1
+            showStep(quize: createStepModel(model: questions[currentQuestionIndex]))
+        }
+    }
+    
+    @IBAction private func noButtonClicked(_ sender: Any) {
+        showAnswerResult(isCorrect: !questions[currentQuestionIndex].correctAnswer)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
```

Comment:
```suggestion
        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
```

По заданию, обновление должно происходить через секунду